.github/workflows/ci.yml

```diff
@@ -0,0 +1,88 @@
+name: CI & Deployment
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  release:
+    types: [published]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: write
+
+env:
+  TSYS_REFERRER: ${{ vars.TSYS_REFERRER }}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: Test and Build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 7
+      - name: 📦 Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      - name: 💨 ESLint
+        run: yarn lint
+      - name: 🧪 Tests
+        run: yarn test
+      - name: ✅ Build
+        run: yarn build
+      - name: Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          fail_ci_if_error: true
+  deploy:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release'
+    name: Deploy to NPM
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 7
+          registry-url: "https://registry.npmjs.org"
+          scope: "CruPCI"
+      - run: yarn install --immutable --immutable-cache
+      - run: yarn publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+  slackNotification:
+    name: Slack Notification
+    needs: [test]
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
Should this be on push or should it be on the actual release going out?

Reply:
I thought it would be on push when the PR gets merged into `master`, but we can make it on `release`.

Reply:
I do not, I've currently got it linked to a channel I created myself, but I'll ask Lee